android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/ActivityResultSender.kt

```diff
@@ -1,7 +1,35 @@
 package com.solana.mobilewalletadapter.clientlib
 
 import android.content.Intent
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.GuardedBy
 
-interface ActivityResultSender {
-    fun launch(intent: Intent)
+class ActivityResultSender(
+    rootActivity: ComponentActivity
```

Comment:
this should be renamed to `activity: Activity`

No need to specifically extend `ComponentActivity` for launching the intent.

Reply:
Kewl, love it

Reply:
Actually, 1ComponentActivity` is required for `registerForActivityResult()` and such